jib-cli/src/test/java/com/google/cloud/tools/jib/cli/WarTest.java

```diff
@@ -0,0 +1,584 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli;
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth8.assertThat;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.tools.jib.api.Credential;
+import com.google.cloud.tools.jib.api.Ports;
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
+import com.google.cloud.tools.jib.cli.logging.HttpTraceLevel;
+import com.google.cloud.tools.jib.cli.logging.Verbosity;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import java.nio.file.Paths;
+import java.time.Instant;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import picocli.CommandLine;
+import picocli.CommandLine.MissingParameterException;
+
+@RunWith(JUnitParamsRunner.class)
+public class WarTest {
```

Comment:
Can you point out which parts have been changed and are worth reviewing in this test?

Reply:
Absolutely. `testParse_defaults()` and `testParse_appRoot()` are the only two methods that have changed.